### PR TITLE
{Keyvault} `az keyvault key create/import/set-attributes/show`: Track2 adoption

### PR DESCRIPTION
**Pull request by evelyn-ys**
**Description**

For SKR(Secure Key Release) feature, we use vendored SDK (See #18089) which is track1 SDK generated privately because:
- Public track1 SDK doesn't update anymore and the latest track1 SDK doesn't cover SKR feature
- `az keyvault key` command group hasn't migrated to track2

After providing service team with edge build using #18089 for private preview, We need to migrate related commands to track2 ASAP

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Keyvault

File: src/azure-cli/setup.py [hunk 1]
```diff
@@ -60,6 +60,7 @@
     'azure-graphrbac~=0.60.0',
     'azure-identity',
     'azure-keyvault-administration==4.0.0b3',
+    'azure-keyvault-keys~=4.4.0',
```
Comment (mccoyp) [inline src/azure-cli/setup.py:63]: Support for SKR was introduced in [`azure-keyvault-keys` 4.5.0b2](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/keyvault/azure-keyvault-keys/CHANGELOG.md#450b2-2021-08-10), and hasn't been included in a non-beta release. Would this need to be updated when track 2 support for SKR is added?

Comment (evelyn-ys) [reply src/azure-cli/setup.py:63]: May I know if `4.5.0b2` APIs are available for all regions and all clouds? CLI will hesitate to use beta SDK if it's not ready.

File: src/azure-cli/azure/cli/command_modules/keyvault/_client_factory.py [hunk 2]
```diff
@@ -205,23 +220,35 @@ def get_token(server, resource, scope):  # pylint: disable=unused-argument
...
+    command_args.pop('vault_base_url', None)
+    command_args.pop('identifier', None)
+    return KeyClient(
+        vault_url=vault_url, credential=credential, api_version=version)
+
+
+def _prepare_data_plane_azure_keyvault_client(cli_ctx, command_args, resource_type):
+    version = str(get_client_api_version(cli_ctx, resource_type))
     profile = Profile(cli_ctx=cli_ctx)
     credential, _, _ = profile.get_login_credentials(resource='https://managedhsm.azure.net')
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/keyvault/_client_factory.py:253]: If keyvault data plane SDK can handle challenge, hard-coding `resource` is unnecessary.

Comment (jiasli) [reply src/azure-cli/azure/cli/command_modules/keyvault/_client_factory.py:253]: `get_raw_token` invocation should be dropped after fully migrating to Track 2,

https://github.com/Azure/azure-cli/blob/5ee89a457012288ad1e542d706f18bc1ae736f0e/src/azure-cli/azure/cli/command_modules/keyvault/_client_factory.py#L156-L157

Comment (evelyn-ys) [reply src/azure-cli/azure/cli/command_modules/keyvault/_client_factory.py]: Removed `resource='https://managedhsm.azure.net'`. Sure, track1 client factory will be deleted after migration. `get_raw_token` will be dropped by then.

**Review by calvinhzy — APPROVED** [review]

**Review by kairu-ms — APPROVED** [review]

**Review by jiasli — APPROVED** [review]

**Pull request merged**